include/RandBLAS/base.hh

```diff
@@ -12,28 +12,35 @@
 #include <typeinfo>
 #include <typeindex>
 
-namespace RandBLAS::base { 
+namespace RandBLAS::base {
 
 #define MIN(a, b) (((a) > (b)) ? (b) : (a))
 #define MAX(a, b) (((a) <= (b)) ? (b) : (a))
 
 enum class RNGName : char {Philox = 'P', Threefry = 'T'};
 
 struct RNGState {
+
     int len_c = 0;
     int len_k = 0;
     uint32_t *ctr = nullptr;
     uint32_t *key = nullptr;
     RNGName rng_name = RNGName::Philox;
 
-    RNGState() {};
+    RNGState() : len_c(0), len_k(0), ctr(nullptr), key(nullptr), rng_name(RNGName::Philox)  {};
```

Comment:
Actually, if we default to Philox then we should set ``len_c = 4`` and ``len_k = 2``. We should also allocate ``ctr`` and ``key`` accordingly.

Reply:
will add enumeration for invalid state.